File: kbs_bbs/bbs2www/src/bbsnot.c
#include "bbslib.h"

int main() {
	FILE *fp;
	char buf[512], board[80], filename[80];
	init_all();
	strsncpy(board, getparm("board"), 32);
	if(!has_read_perm(currentuser, board)) http_fatal("错误的版面");
	printf("<center>\n");
	printf("%s -- 备忘录 [讨论区: %s]<hr color=green>\n", BBSNAME, board);
   	sprintf(filename, "vote/%s/notes", board);
	fp=fopen(filename, "r");
	if(fp==0) {
		printf("<br>本讨论区尚无「进版画面」。\n");
		http_quit();
	}
   	printf("<table border=1 width=610><tr><td><pre>\n");
   	while(1) {
		char *s;
		bzero(buf, 512);
		if(fgets(buf, 512, fp)==0) break;
		while(1) {
			int i;
			s=strstr(buf, "$userid");
			if(s==0) break;
			for(i=0; i<7; i++) s[i]=32;
			for(i=0; i<strlen(currentuser->userid); i++)
				s[i]=currentuser->userid[i];
		}
		hhprintf("%s", buf);
	}
 	fclose(fp); 
 	printf("</pre></table><hr>\n");
   	printf("[<a href=bbsdoc?board=%s>本讨论区</a>] ", board);
	if(has_BM_perm(currentuser, board)) 
		printf("[<a href=bbsmnote?board=%s>编辑进版画面</a>]", board);
   	printf("</center>\n"); 
	http_quit();
}

File: kbs_bbs/bbs2www/html/bbsnot.php
<?php
	/**
	 * $Id$ 
	 */
	require("funcs.php");
	if ($loginok !=1 )
		html_nologin();
	else
	{
                $usernum = $currentuser["index"];
		if (isset($_GET["board"]))
			$board = $_GET["board"];
		else 
			html_error_quit("错误的讨论区");
                $brdarr = array();
                $brdnum = bbs_getboard($board, $brdarr);
                if ($brdnum == 0)
			html_error_quit("错误的讨论区");
		if (bbs_checkreadperm($usernum,$brdnum)==0)
			html_error_quit("错误的讨论区");
		$top_file="vote/" . $board . "/notes";
		$fp = fopen($top_file, "r");
		if ($fp == FALSE) {
		        html_init("gb2312");
			html_error_quit("现在没有备忘录");
                }
                fclose($fp);
                if (cache_header("public",filemtime($top_file),300))
                	return;
		html_init("gb2312");
?>
<body>
<center><?php echo $BBS_FULL_NAME; ?> -- 备忘录 [讨论区: <?php echo $board; ?>]<hr color=green>
<table border=1 width=610><tr><td><pre>
<?php
	bbs_printansifile($top_file);
?></tr></td>
</pre></table>
[<a href=bbsdoc.php?board=<?php echo $board; ?>>本讨论区</a>]<?
    if (bbs_is_bm($brdnum,$usernum))
	echo "[<a href=bbsmnote.php?board=" . $board . ">编辑进版画面</a>]";
?> 
</center>
<?php
		html_normal_quit();
	}
?>
